src/client_side.cc

```diff
@@ -3953,9 +3950,13 @@ ConnStateData::terminateAll(const Error &rawError, const LogTagsErrors &lte)
     // may be set during the pipeline cleanup above
     connLeftovers_ = false;
 
-    if (intputToConsume && !inBuf.isEmpty()) {
-        debugs(83, 5, "forgetting client " << intputToConsume << " bytes: " << inBuf.length());
-        inBuf.clear();
+    if (!inBuf.isEmpty()) {
+        if (intputToConsume) {
+            debugs(83, 5, "forgetting client " << intputToConsume << " bytes: " << inBuf.length());
+            inBuf.clear();
+        } else {
+            bareError.update(error ? error : ERR_STREAM_FAILURE); // XXX: bareLogTagsErrors
```

Comment:
This change secretly drops WITH_CLIENT detail for ERR_NONE cases (those cases are converted to ERR_STREAM_FAILURE here). Assuming that drop is intentional/correct, we should adjust this function code (including comments) to make that drop explicit/expected.

Reply:
Adjusted. I left the XXX for now - I could not recollect its meaning.